OneCommunityGlobal/HighestGoodNetworkApp
# Yiyun create page for same folder tasks - first part

This PR is the frontend changes for implementing "see tasks in the same folder as the current one". This PR is only the first half part, it just implement when the current task is level one task, it will lead to the WBS page contains the current task. The second half part - implements see same folder tasks as the level2/3/4 tasks, will come in the next PR.

Mainly changes explained:
1. `src/components/Projects/WBS/SameFolderTasks/SameFolderTasks.js` - create the `SameFolderTask` component for page display, the condition is for whether the current task is level 1 or level 2/3/4 task
2. src/components/Projects/WBS/SingleTask/SingleTask.js - create the different views for different role users, only owner/admin can see this feature

Test steps:
To test this changes, you would need also get the [backend changes](https://github.com/OneCommunityGlobal/HGNRest/pull/183). Then click nav icon on top left on singleTask page.
If the current task is a level 1 task, it would lead to below image 2; if it's a leve 2/3/4 task, it would lead to below image 3, which is still uncompleted now.

<img width="961" alt="image" src="https://user-images.githubusercontent.com/5071040/186774120-b8957ebc-cd6f-4fe9-b8bf-e5df742edfa8.png">
<img width="947" alt="image" src="https://user-images.githubusercontent.com/5071040/186774261-3958e565-a78a-4eb4-bb2a-9926ba5986d0.png">
<img width="427" alt="image" src="https://user-images.githubusercontent.com/5071040/186774562-81370691-b1c4-42e5-8efa-c4b94f05f330.png">
 
Potentially bug: ❓ 
On the sameFolderTasks page, it should look like the below image, but sometime the highlight part is missing. That because on line 32 `src/components/Projects/WBS/SameFolderTasks/SameFolderTasks.js`, the`WBS` need sometime to get data instead of the initial empty object. I'm guessing I used two `useEffect()`s there so somehow it delays on getting data. 
If the bug cannot get solved in this PR, I will document it on our Management Dashboard bug list. 
<img width="770" alt="image" src="https://user-images.githubusercontent.com/5071040/186774924-ccfdba28-0290-45a2-a2ec-6a265675ddbf.png">
